### PR TITLE
Presto properties description

### PR DESCRIPTION
**Pull request by fiedukow**
This is continuation of https://github.com/prestodb/presto/pull/4987, moved here as 250 comments is to much for github to handle smoothly.

Properties description was moved to separate article.

@electrum @dain @KBP-TDC

### CODE REVIEW

**Comment by KBP-TDC** [comment]
LGTM

File: presto-docs/src/main/sphinx/admin/properties.rst [hunk 1]
```diff
@@ -0,0 +1,487 @@
...
+ * **Default value:** ``true``
+ * **Description:** Use hash distributed joins instead of broadcast joins. Distributed joins require redistributing both tables using a hash of the join key. This can be slower (sometimes substantially) than broadcast joins but allows much larger joins. Broadcast joins require that the tables on the right side of the join after filtering fit in memory on each node whereas distributed joins only need to fit in distributed memory across all nodes. This can also be specified on a per-query basis using the ``distributed_join`` session property.
+
+
+``redistribute-writes``
+^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``true``
+ * **Description:** Force parallel distributed writes. Setting this property will cause the write operator to be distributed between nodes. This allows allows Presto to more efficiently utilize distributed storage across the cluster. It is particularly effective when executing a small number of huge queries. This can also be specified on a per-query basis using the ``redistribute_writes`` session property.
```
Comment (pnowojski) [inline presto-docs/src/main/sphinx/admin/properties.rst]: Actually I think it doesn't work that way.

This turns on data redistribution between nodes before writing them to table writers. It's helpful if there is an significant data skew in `SOME_SKEWED_QUERY` when it is used in:

`INSERT INTO TABLE xyz SELECT SOME_SKEWED_QUERY`
`CREATE TABLE xyz AS SELECT SOME_SKEWED_QUERY`

File: presto-docs/src/main/sphinx/admin/properties.rst [hunk 2]
```diff
@@ -0,0 +1,487 @@
...
+
+Node scheduler properties
+-------------------------
+
+``node-scheduler.max-pending-splits-per-node-per-stage``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer``
+ * **Default value:** ``10``
+ * **Description:** Must be smaller than ``node-scheduler.max-splits-per-node``. This property describes how many splits can be queued to each worker node. Having this value higher will allow more jobs to be queued but will cause resources to be used for that. Using a higher value is recommended if queries are submitted in large batches, (eg. running a large group of reports periodically). Increasing this value may help to avoid query drops and decrease the risk of short query starvation. To high value may drastically increase processing wall time if node distribution of query work will be skew. This is especially important if nodes do have important differences in performance. The best value for that is enough to provide at least one split always waiting to be process but not higher.
```
Comment (pnowojski) [inline presto-docs/src/main/sphinx/admin/properties.rst]: ` To high value` ->  `Too high value`

File: presto-docs/src/main/sphinx/admin/properties.rst (hunk 1 above)
Comment (fiedukow) [inline presto-docs/src/main/sphinx/admin/properties.rst]: Fixed.

File: presto-docs/src/main/sphinx/admin/properties.rst (hunk 2 above)
Comment (fiedukow) [inline presto-docs/src/main/sphinx/admin/properties.rst]: Fixed.

**Comment by fiedukow** [comment]
Rebased to master, conflicts solved.

**Comment by fiedukow** [comment]
Can we decide what to do with that, it hangs in same state from 16 Jul.
@electrum

**Comment by electrum** [comment]
I started working on this, but am maybe 1/3 of the way through. It's going to take a lot more work to complete.

**Comment by fiedukow** [comment]
Hey there.
Any new notes?
Maybe you have something to share from the part that you already reviewed?

**Comment by fiedukow** [comment]
Kind ping :-)

**Comment by fiedukow** [comment]
Rebased to master, kind ping again :)

**Comment by electrum** [comment]
Sorry for the long delay on this. In order to get this in, can we break this up into multiple commits:

* Hive properties
* One commit per section of general tuning properties (I counted 7)

Just put them in order in this PR. I can review each one relatively quickly (hour or two) and merge them one by one.

**Comment by fiedukow** [comment]
Done. I also separated commits related to old descriptions, to show exactly what has changed.

**Comment by fiedukow** [comment]
Any progress or comments for those new commits?

**Comment by electrum** [comment]
Sorry, missed your note that this was updated. Will review this week.

**Comment by fiedukow** [comment]
Hm, any luck with finding some time for that? :)

**Comment by fiedukow** [comment]
kind ping again

**Comment by fiedukow** [comment]
image
Not that I would like to ping again, but we have a little anniversary to celebrate today :-D

**Comment by maciejgrzybek** [comment]
Nice Presto logo made of caramel! :D
We should have such a cake on the next meetup! :D

**Comment by electrum** [comment]
- [ ]   d84ed4996     Migrate properties description to separate doc page
- [ ]   2cee27c73     Adjust properties doc page to fit new format
- [ ]   84c408b9f     Move general properties to the begining of properties doc
- [ ]   d1795204d     Fix and extend exisitng properties description
- [ ]   296f11605     Add general properties description
- [ ]   db68c3140     Add exchange properties description
- [ ]   b0487f411     Add node scheduler properties description
- [ ]   4e4ea31aa     Add optimizer properties description
- [ ]   61f00540a     Add query properties description
- [ ]   9294919db     Add task properties description
- [ ]   e222a6f07     Add session properties description
- [ ]   a8153933f     Add more detailed description of Hive connector properties
- [ ]   dfce20147     Add regular expression properties description
- [ ]   6e6ec8194     Add information about most important performance releated properties
- [ ]   fedc15963     Add tuning for queries with fast processing splits
- [ ]   55d0bd52c     Add spilling properties documentation

**Comment by kokosing** [comment]
@electrum Can you please confirm that marked commits are ready to go? If so I could extract them as separate PR and merge independently?

**Comment by electrum** [comment]
All of the marked commits have been merged (with editing).

**Comment by kokosing** [comment]
Thank you. I will try to rebase this then (in my spare time).

**Comment by anusudarsan** [comment]
Closing this in favor of https://github.com/prestodb/presto/pull/9439 (PR created with the unmerged changes)

**Pull request closed without merging**